common/include/pcl/experimental/executor/property/blocking.hpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception, Inc.
+ *  Author: Shrijit Singh <shrijitsingh99@gmail.com>
+ *
+ */
+
+#pragma once
+
+#include <pcl/experimental/executor/property/base_property.hpp>
+
+namespace executor {
+
+/**
+ * \brief A behavioral property that guarantees executors provide about the
+ * blocking behavior of their execution functions.
```

Comment:
(for a reader) What's blocking behavior?

Reply:
Added reference to the proposal. I will touch upon this on the executor design document.

Reply:
Some explanation of "what blocking is" here would also be helpful

Reply:
Added a one line explanation